### PR TITLE
Preserve inputs and outputs order for TensorFlow 2 and MXNet models

### PR DESCRIPTION
**Pull request by popovaan**
Description: Implemented preserving of input/output order for TF2 and MxNet.
Depends on https://github.com/openvinotoolkit/openvino/pull/9352

Ticket: CVS 64960

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR - N/A
* [x]  Transformation preserves original framework node names - N/A

Validation:
* [x]  Unit tests
* [x]  Framework operation tests - will be added in separate PR.
* [x]  Transformation tests - N/A
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list - N/A
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

### CODE REVIEW

**Comment by achetver** [comment]
In general looks good.

@popovaan have you checked these changes on real models?

**Review by vgavrilo — APPROVED** [review]

**Review by achetver — APPROVED** [review]

**Comment by rkazants** [comment]
> > In general looks good.
> > @popovaan have you checked these changes on real models?
> 
> Yes, I have generated few models with multiple input/output for TF2 and MxNet and checked PR on them.

It would be great if you share some code (TF2, MXNet and OpenVINO) and demonstrate how we preserve their order. Lists of inputs and outputs are the same among these frameworks.

File: tools/mo/openvino/tools/mo/front/tf/loader.py [hunk 1]
```diff
@@ -236,7 +244,21 @@ def load_tf_graph_def(graph_file_name: str = "", is_binary: bool = True, checkpo
                 graph_def = frozen_func.graph.as_graph_def(add_shapes=True)
                 # disable eager execution since next steps are executed with a graph in non-eager mode
                 tf_v1.disable_eager_execution()
-                return graph_def, variables_values, 'tf2'
+
+                input_names = []
+                if hasattr(imported, 'inputs'):
+                    # Extract tensor names order from Keras model
+                    input_names = [tensor.name for tensor in imported.inputs]
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/loader.py:251]: why don't we extract input tensor names by using `frozen_func.inputs`. It should also have the same order of inputs.
So you don' t need new code above for TF for loading

Comment (popovaan) [reply tools/mo/openvino/tools/mo/front/tf/loader.py:251]: No, order is not the same. Inputs/outputs in frozen_func are sorted by alphabet, which is not correspond original model. But during model inference the order should be as in original model, not sorted. I checked it in on layer tests. 
frozen_func does not have original order, the order is stored only in saved_model format for TF2. But the only problem is that tensor names of outputs are renamed after freezing (input names are not renamed). For this reason I used frozen_func.outputs which are sorted by "Identity" names.

File: tools/mo/openvino/tools/mo/front/tf/loader.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 from openvino.tools.mo.utils.error import Error, FrameworkError
 from openvino.tools.mo.utils.utils import refer_to_faq_msg
 from openvino.tools.mo.utils.versions_checker import get_environment_setup
+from tensorflow.python.tools import saved_model_utils
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/loader.py:13]: it looks nowhere used routine. Please remove

**Review by rkazants — COMMENTED** [review]
please fix one last comment. The other stuff looks good

File: tools/mo/openvino/tools/mo/front/tf/loader.py (hunk 2 above)
Comment (popovaan) [reply tools/mo/openvino/tools/mo/front/tf/loader.py]: Done.

Comment (popovaan) [reply tools/mo/openvino/tools/mo/front/tf/loader.py]: Done.

**Review by rkazants — APPROVED** [review]

**Pull request merged**